netann/chan_status_manager.go

```diff
@@ -263,6 +268,13 @@ func (m *ChanStatusManager) RequestDisable(outpoint wire.OutPoint,
 	return m.submitRequest(m.disableRequests, outpoint, manual)
 }
 
+// RequestAuto submits a request to restore automatic channel state management.
+// If the channel is in the state ChanStatusManuallyDisabled, it will be moved
+// back to the state ChanStatusDisabled. Otherwise, no action will be taken.
+func (m *ChanStatusManager) RequestAuto(outpoint wire.OutPoint) error {
```

Comment:
Thanks! The semantics are a bit confusing since there are so many interactions, and the behavior is different between manual enable and manual disable, but hopefully this is a reasonable approach. (Happy to revisit if needed though!)